### ISSUE TITLE
how to disable this widget for some figures?

### ISSUE THREAD

**Issue by HYuanggg** [matplotlib/ipympl#219]
as title. Thank you in advance.

**Comment by thomasaarholt** [comment]
Can we assume that you mean, "and use the inline backend instead"? :) As in, use `%matplotlib inline` as the backend for some figures?

**Comment by HYuanggg** [comment]
> Can we assume that you mean, "and use the inline backend instead"? :) As in, use `%matplotlib inline` as the backend for some figures?

I tried '%matplotlib inline' and it is like not iteractive, which is what I want. But I think I need to add '%matplotlib inline' or '%matplotlib widget' every time I want to change from one to the other, right?

**Comment by martinRenou** [comment]
Yeah that would be it.. There is no way, I think, to do otherwise.

**Comment by HYuanggg** [comment]
> Yeah that would be it.. There is no way, I suppose, to do otherwise.

Thank you Martin

**Comment by thomasaarholt** [comment]
This issue is a bit more general than ipympl, I think. I've wondered a lot about the general matplotlib backend implementation in jupyter - like how one cannot switch between the various interactive backends without restarting the kernel, for instance.

**Comment by martinRenou** [comment]
This is completely dependent to ipython actually, ipython implements the magics that decide how matplotlib behaves.

**Comment by ianhi** [comment]
Does the below achieve what you want?
```python
plt.ioff(); fig = plt.figure(); plt.ioff()
# whatever plotting you want
display(fig)
```

Using ioff prevents the automatic display of the figure on it's creation, and then if you `display(fig)` instead of `display(fig.canvas)` you will get a png as output like the inline backend.

**Comment by hckr** [comment]
@ianhi I think you didn't mean to write `plt.ioff()` the second time, but rather `plt.ion()`?

The following works for me:
```python
with plt.ioff():
    fig = plt.figure()
    # ...
    display(fig)
```

**Comment by ianhi** [comment]
> @ianhi I think you didn't mean to write plt.ioff() the second time, but rather plt.ion()?

good catch!. I edited it to be the correct thing

**Comment by Fernandogarciagoatcoder** [comment]
For me even whith the above code ( with plt.ioff()...etc) I still get the interactive figure and right below the non-interactive one. Is there a way to completelty remove the interactive figure ?
